src/linear_algebra/matrix.lean

```diff
@@ -813,6 +820,32 @@ begin
   apply h₂,
 end
 
+variables {V : Type*} [add_comm_group V] [vector_space K V]
+variables {v₁ : n → V} (hv₁ : is_basis K v₁)
+
+lemma ker_to_lin_eq_bot [decidable_eq n] (M : matrix n n K) (hM : M.det ≠ 0) :
+  (to_lin hv₁ hv₁ M).ker = ⊥ :=
+begin
+  rw submodule.eq_bot_iff,
+  intros x hx,
+  calc x = to_lin hv₁ hv₁ (M⁻¹ ⬝ M) x : by simp [nonsing_inv_mul M (is_unit.mk0 _ hM)]
+     ... = to_lin hv₁ hv₁ M⁻¹ (to_lin hv₁ hv₁ M x) : by simp [to_lin_mul hv₁ hv₁ hv₁]
+     ... = (to_lin hv₁ hv₁ M⁻¹) 0 : by rw mem_ker.mp hx
+     ... = 0 : linear_map.map_zero _
+end
+
+lemma range_to_lin_eq_top [decidable_eq n] (M : matrix n n K) (hM : M.det ≠ 0) :
+  (to_lin hv₁ hv₁ M).range = ⊤ :=
+begin
+  rw eq_top_iff,
+  rintros x -,
+  rw linear_map.mem_range,
+  use to_lin hv₁ hv₁ M⁻¹ x,
+  rw [← linear_map.comp_apply, ← to_lin_mul, mul_nonsing_inv _ (is_unit.mk0 _ hM),
+      to_lin_one, linear_map.id_apply]
+end
```

Comment:
I think it would be cool to combine these into
```
def to_lin_equiv (M : matrix n n K) (hM : M.det ≠ 0) : V ≃ₗ[R] V
```
I suspect you can construct that equiv directly with a shorter proof than your current pair